Source/Dafny/Resolver.cs

```diff
@@ -14195,7 +14201,9 @@ public class ResolveOpts
 
         ResolveExpression(e.Array, opts);
         Contract.Assert(e.Array.Type != null);  // follows from postcondition of ResolveExpression
-        Type elementType = new InferredTypeProxy();
+        Type elementType = e.Array.Type.TypeArgs.Count > 0 ?
```

Comment:
Can `e.Array.Type.TypeArgs` ever be null? Wanted to double check that we don't need a 
`Contract.Assert(e.Array.Type.TypeArgs != null);` as well (or something similar).

Reply:
It appears that it is always non-null, with at least a TypeProxy, at least for all the lit tests; I added a test where a variable was just declared with type array2, but it still is not null. Added a Contract.Assert rather than a branch of code.